plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -179,7 +179,7 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Update
             },
                 e -> actionListener
                     .onFailure(new MLResourceNotFoundException("Failed to find model to update with the provided model id: " + modelId))
-            ), () -> context.restore()));
+            ));
```

Comment:
Why remove this?

Reply:
We've restored the context in line 488, if we restore here it will cause big

Reply:
Suggest keep restore here. Then add stash context before running update request.